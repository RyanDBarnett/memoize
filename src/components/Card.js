import React from 'react';
import '../styles/Card.css';
import generateChoices from '../utils/generateChoices';
import endRoundBtns from '../utils/endRoundBtns';

const Card = ({
  currentQuestion,
  showNextQuestion,
  resetAllQuestions,
  resetIncorrectQuestions,
  checkAnswer
  }) => {
  const choices = currentQuestion && currentQuestion.choices;
  const buttons = currentQuestion ? generateChoices(choices, showNextQuestion, checkAnswer) : endRoundBtns(resetAllQuestions, resetIncorrectQuestions);
  const display = currentQuestion ? <h4>{currentQuestion.question}</h4> : <h4>All Done! Do you want to go again?</h4>
  return (
    <div className="Card">
      {display}
      {buttons}
    </div>
  )
}

export default Card;