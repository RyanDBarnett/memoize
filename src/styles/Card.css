.Card {
  width: 35%;
  z-index: 0;
  background-color: #e3f0f6;
  padding: 2%;
  border-radius: 20px;
  border: #211a90 2px solid;
  text-align: center;
}

.Card h4 {
  min-height: 75px;
}
