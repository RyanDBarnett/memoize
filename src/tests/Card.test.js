import React from 'react';
import Card from '../components/Card';
import { shallow } from "enzyme";

describe('Card', () => {
  let wrapper;

  const mockShowNextQuestion = jest.fn()
  const mockResetAllQuestions = jest.fn()
  const mockResetIncorrectQuestions = jest.fn()
  const mockCheckAnswer = jest.fn()

  const mockCurrentQuestion = {
    answer: "Roles",
    choices: ["Properties","States","Roles"],
    id: "1",
    question: "Which of the 3 ARIA main features defines what an element is or does?"
  }

  beforeEach( () => {
    wrapper = shallow(
      <Card 
        currentQuestion={mockCurrentQuestion}
        showNextQuestion={mockShowNextQuestion}
        resetAllQuestions={mockResetAllQuestions}
        resetIncorrectQuestions={mockResetIncorrectQuestions}
        checkAnswer={mockCheckAnswer}
      />
    )
  });
  
  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should display ending buttons and heading if there are no more cards", () => {
    wrapper = shallow(
      <Card 
        currentQuestion={null}
      />
    )
    expect(wrapper).toMatchSnapshot();
  });
});