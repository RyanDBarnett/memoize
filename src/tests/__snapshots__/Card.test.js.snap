// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should display ending card if there are no more cards 1`] = `
<div
  className="Card"
>
  <h4>
    All Done! Do you want to go again?
  </h4>
  <div
    className="btnContainer"
  >
    <button
      className="resetAll"
    >
      Reset All Questions
    </button>
    <button
      className="resetIncorrect"
    >
      Practice Incorrect Questions
    </button>
  </div>
</div>
`;

exports[`Card should match the snapshot with all data passed in 1`] = `
<div
  className="Card"
>
  <h4>
    Which of the 3 ARIA main features defines what an element is or does?
  </h4>
  <div
    className="btnContainer"
  >
    <button
      className="choice-btn-1"
      key="0"
      onClick={[Function]}
    >
      Properties
    </button>
    <button
      className="choice-btn-2"
      key="1"
      onClick={[Function]}
    >
      States
    </button>
    <button
      className="choice-btn-3"
      key="2"
      onClick={[Function]}
    >
      Roles
    </button>
  </div>
</div>
`;
